frontend/src/app/services/http.service.ts

```diff
@@ -19,7 +19,7 @@ export class HttpService {
 
   getToken(username: string, password: string): Observable<LoginResponse> {
     if (username.length < minLength || password.length < minLength) {
-      return throwError(`HTTP_SERVICE: username or password too short`);
+      return throwError({error: `Username or password too short`});
```

Comment:
It is a good practice to implement a form-based validation which is usually the first error filter:
https://angular.io/guide/form-validation

Reply:
I'll take a look into it ;)